packages/share_plus/share_plus_web/lib/share_plus_web.dart

```diff
@@ -61,7 +62,107 @@ class SharePlusPlugin extends SharePlatform {
     String? subject,
     String? text,
     Rect? sharePositionOrigin,
-  }) {
-    throw UnimplementedError('shareFiles() has not been implemented on Web.');
+  }) async {
+    final files = <XFile>[];
+    for (var i = 0; i < paths.length; i++) {
+      files.add(XFile(paths[i], mimeType: mimeTypes?[i]));
+    }
+    return shareXFiles(
+      files,
+      subject: subject,
+      text: text,
+      sharePositionOrigin: sharePositionOrigin,
+    );
+  }
+
+  /// Share [XFile] objects.
+  ///
+  /// Remarks for the web implementation:
+  /// This uses the [Web Share API](https://web.dev/web-share/) if it's
+  /// available. Otherwise it falls back to downloading the shared files.
+  /// See [Can I Use - Web Share API](https://caniuse.com/web-share) to
+  /// understand which browsers are supported. This builds on the
+  /// [`cross_file`](https://pub.dev/packages/cross_file) package.
+  @override
+  Future<void> shareXFiles(
+    List<XFile> files, {
+    String? subject,
+    String? text,
+    Rect? sharePositionOrigin,
+  }) async {
+    // See https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share
+
+    final webFiles = <html.File>[];
+    for (final xFile in files) {
+      webFiles.add(await _fromXFile(xFile));
+    }
+    try {
+      await _navigator.share({
+        if (subject?.isNotEmpty ?? false) 'title': subject,
+        if (text?.isNotEmpty ?? false) 'text': text,
+        if (webFiles.isNotEmpty) 'files': webFiles,
+      });
+    } on NoSuchMethodError catch (exception, stackTrace) {
+      FlutterError.onError?.call(FlutterErrorDetails(
+        exception: exception,
+        stack: stackTrace,
+        context: DiagnosticsNode.message('while trying to share file(s)'),
+        library: 'share_plus_web',
+        informationCollector: () => [
+          DiagnosticsNode.message(
+            'The web share API is not available on the current browser. '
+            'Falling back to downloading the files. '
+            'Subject and text are discarded if given.',
+          ),
+        ],
+      ));
```

Comment:
Should we call directly into the FlutterError error sink? 

I think it would be better to throw an exception instead and let the developer catch it on their side.

Reply:
Done, the exceptions are now uncaught